reana_server/complexity.py

```diff
@@ -0,0 +1,356 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2021 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""REANA workflow complexity estimation."""
+
+import os
+from functools import reduce
+
+from reana_commons.job_utils import kubernetes_memory_to_bytes
+from reana_commons.config import MQ_MAX_PRIORITY
+
+from reana_server.status import NodesStatus
+from reana_server.config import REANA_KUBERNETES_JOBS_MEMORY_LIMIT
+
+
+def calculate_workflow_priority(complexity):
+    """Calculate workflow priority based on its complexity."""
+
+    def get_bin(memory):
+        for b in range(bins):
+            if memory in range(bin_size * b, bin_size * (b + 1)):
+                return b
+        return bins - 1
+
+    # Summing up parrarel jobs times memory to determine the workflow complexity
+    memory = int(reduce(lambda sum, item: sum + item[0] * item[1], complexity, 0))
+    bins = MQ_MAX_PRIORITY
+    total_cluster_memory = NodesStatus().get_total_memory()
+    bin_size = int(total_cluster_memory / bins)
+    priority = bins - get_bin(memory)
+    return priority
```

Comment:
Isn't it enough to do something like this?
```python
int((1 - wf_memory / total_cluster_memory) * MQ_MAX_PRIORITY)
```

Maybe I'm missing something.

Reply:
Well spotted, indeed this gives the same result, but in a much nicer way